### PR TITLE
DAOS-623 review: Enable vos-watcher in CODEOWNERS

### PR DESCRIPTION
**Pull request by jolivier23**
Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -7,11 +7,12 @@
 # any PR that touches Go files should get a review from go-owners
 #*.go @daos-stack/go-owners
 
-# any PR that touches VOS files should get a review from VOS
-#src/vos/ @daos-stack/vos
-
-# any PR that touches btree files should get a review from VOS
-#src/common/btree*.* @daos-stack/vos
+# Notify vos-watcher of files touched affecting VOS
+src/vos/ @daos-stack/vos-watcher
```
Comment (mjmac) [inline .github/CODEOWNERS:11]: It's a nitpick, but it seems like these team names should be pluralized. e.g. vos-watchers, go-watchers, etc.

Would you mind making that update and also adding go-watchers to the line above this one?

Comment (jolivier23) [reply .github/CODEOWNERS:11]: @mchaarawi would need to update his new team as well

Comment (jolivier23) [reply .github/CODEOWNERS:11]: Perhaps, we should just merge his PR into this one

Comment (jolivier23) [reply .github/CODEOWNERS:11]: I'll update them all

**Pull request merged**